fixtures/flight-vite/server/global.js

```diff
@@ -0,0 +1,259 @@
+'use strict';
+
+// This is a server to host CDN distributed resources like module source files and SSR
+
+const path = require('path');
+const url = require('url');
+
+const fs = require('fs').promises;
+const compress = require('compression');
+const chalk = require('chalk');
+const express = require('express');
+const http = require('http');
+const vite = require('vite');
+
+const {renderToPipeableStream} = require('react-dom/server');
+const {createFromNodeStream} = require('react-server-dom-vite/client');
+
+const moduleBasePath = new URL('../src', url.pathToFileURL(__filename)).href;
+
+async function createApp() {
+  const app = express();
+
+  app.use(compress());
+
+  function request(options, body) {
+    return new Promise((resolve, reject) => {
+      const req = http.request(options, res => {
+        resolve(res);
+      });
+      req.on('error', e => {
+        reject(e);
+      });
+      body.pipe(req);
+    });
+  }
+
+  let getClientAsset;
+
+  if (process.env.NODE_ENV === 'development') {
+    const vite = await import('vite');
+    const {default: reactRefresh} = await import('@vitejs/plugin-react');
+    const viteServer = await vite.createServer({
+      appType: 'custom',
+      server: {middlewareMode: true},
+      plugins: [
+        reactRefresh(),
+        {
+          name: 'react-server-dom-vite:react-refresh',
+          handleHotUpdate({file}) {
+            // clear vite module cache so when its imported again, we will
+            // get the new version
+            globalThis.__vite_module_cache__.delete(file);
+          },
+        },
+      ],
+      ssr: {
+        external: ['react', 'react-dom', 'react-server-dom-vite'],
+      },
+    });
+
+    globalThis.__vite_module_cache__ = new Map();
+    globalThis.__vite_preload__ = metadata => {
```

Comment:
I would think that you would use the unbundled version for the fixture or maybe have two fixtures one for unbundled and one for running vite on the server.

I'm not very familiar with vite though, does vite do a compilation to replace magic globals like __vite_preload__ ? or does it always read them from the global scope?

I would assume you'd want to have the fixture here do whatever a real production vite project would do to make sure it is asserting real behavior